plugins/chat/assets/javascripts/discourse/components/chat-message.js

```diff
@@ -104,6 +110,9 @@ export default class ChatMessage extends Component {
     };
 
     this.args.message.expanded = true;
+    schedule("afterRender", () => {
```

Comment:
I would most likely have the after render directly in `refreshStatusOnMentions` function, you shouldn't need to have the knowledge that this function needs to be wrapped in an after render block

Reply:
I think, more precisely, this `schedule("afterRender"` isn't related to `refreshStatusOnMentions`, it's related to the line above:

`this.args.message.expanded = true;`

At this moment, the message is collapsed, we're expanding it by setting `expanded` to `true`. If we want to do something with the expanded message in UI right away (it can be a call to `refreshStatusOnMentions` or it can be something else), we have to schedule that to the afterRender queue.

Other case when we call `refreshStatusOnMentions` is from modifiers, and in those cases there is no need to schedule it to the after render queue.

I feel, from this point of view, this is a good place for this call?

Reply:
I see it the opposite way, the code in `refreshStatusOnMentions` needs to act on the DOM, as a result we need to always be sure we are not in the middle of changing the DOM. This is always going to be related to something outside of `refreshStatusOnMentions`, but leaving the responsibility to the developer to call schedule before calling it means you need to know the implementation of `refreshStatusOnMentions` and even details of it, before calling it.